File: python/Majority Element.py
class Solution(object):
    def majorityElement(self, nums):
        votes = 0
        number = 0
        for num in nums:
            if number == num:
                votes += 1
            elif votes == 0:
                number = num
                votes += 1
            else:
                votes -= 1
        return number

File: cpp/Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        if (nums.size() == 0) {
            return 0;
        }
        int majority = nums[0], count = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == majority) {
                count++;
            } else {
                count--;
                if (count < 0) {
                    majority = nums[i];
                    count = 0;
                }
            }
        }
        return majority;
    }
};
